### PR TITLE
Add hotkeys for incrementing/decrementing select state slot

### PR DESCRIPTION
**Pull request by askew-etc**
RetroArch has hotkeys you can assign to increment/decrement the currently selected save state slot. This is convenient for controller users since you can change the save state slot and then save to or load from different slots without having to touch the keyboard. This functionality could be desirable in Dolphin.

### CODE REVIEW

File: Source/Core/DolphinQt/MainWindow.cpp [hunk 1]
```diff
@@ -1358,6 +1362,30 @@ void MainWindow::SetStateSlot(int slot)
...
+void MainWindow::IncrementSelectStateSlot()
+{
+  int state_slot = Settings::Instance().GetStateSlot();
+  state_slot++;
+  if (state_slot > 10)
+  {
+    state_slot = 1;
+  }
+  m_state_slot = state_slot;
+  m_menu_bar->SetStateSlot(m_state_slot);
```
Comment (AdmiralCurtiss) [inline Source/Core/DolphinQt/MainWindow.cpp:1374]: I think you're overcomplicating this, it should be enough to just do

```
  int state_slot = m_state_slot + 1;
  if (state_slot > 10)
    state_slot = 1;
  SetStateSlot(state_slot);
```

(similar for the decrement case)

Comment (AdmiralCurtiss) [reply Source/Core/DolphinQt/MainWindow.cpp:1374]: You probably should also use `State::NUM_STATES` instead of `10` for the max state number.

Comment (askew-etc) [reply Source/Core/DolphinQt/MainWindow.cpp]: Sounds good. I made the changes to the code per your recommendations.

**Comment by Pokechu22** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/HotkeyManager.cpp [hunk 2]
```diff
@@ -179,6 +179,8 @@ constexpr std::array<const char*, NUM_HOTKEYS> s_hotkey_labels{{
     _trans("Undo Save State"),
     _trans("Save State"),
     _trans("Load State"),
+    _trans("Increase Select State Slot"),
+    _trans("Decrease Select State Slot"),
```
Comment (Pokechu22) [inline Source/Core/Core/HotkeyManager.cpp:183]: Grammatically, I think these should be "Increase Select<ins>ed</ins> State Slot"/"Decrease Select<ins>ed</ins> State Slot" (and the same for `HK_INCREMENT_SELECT_STATE_SLOT`/`IncrementSelectStateSlotHotkey`/`IncrementSelectStateSlot` and the decrement ones).

Comment (askew-etc) [reply Source/Core/Core/HotkeyManager.cpp]: Ok, I've made the changes you requested.

**Comment by Pokechu22** [comment]
@dolphin-emu-bot rebuild

**Comment by AdmiralCurtiss** [comment]
You should squash the fixup commits but otherwise this looks good to me.

Your first commit also has a weird author that's not associated with any GitHub account.

**Comment by AdmiralCurtiss** [comment]
@dolphin-emu-bot rebuild

**Review by AdmiralCurtiss — APPROVED** [review]
Untested.

**Comment by AdmiralCurtiss** [comment]
Seems to work correctly, too.

**Pull request merged**